cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/LocalManager.java

```diff
@@ -107,6 +111,7 @@ public LocalManager(CConfiguration configuration, PipelineFactory pipelineFactor
     pipeline.addLast(new DeployDatasetModulesStage(datasetFramework));
     pipeline.addLast(new CreateDatasetInstancesStage(datasetFramework));
     pipeline.addLast(new CreateStreamsStage(id, streamAdmin, exploreFacade, exploreEnabled));
+    pipeline.addLast(new CreateSchedulesStage(store, scheduler));
```

Comment:
I think this is a weakness in the way that the deploy pipeline is handled. Or perhaps not. Because every stage can pass to the next stage whatever it needs to. So one could have a stage that takes a snapshot of the existing state (app spec, schedules, whether they are active, ...) and pass it into the pipeline. Then every stage has to pass it on from there on. When you reach the schedule stage, that old information would be in the pipeline, whereas the new app spec is already in the store (or also in the pipeline). To be honest, I am not sure whether the pipeline is designed well. Feels like every stage should be able to add information to the state that is passed between stages, without having to change the next stage. So perhaps what is passed on should be a map (key->Object), and every stage only extracts the things from the map that it needs, and can add new things that downstream stages might need.